src/lib/utils/settings/abstract_setting.hpp

```diff
@@ -30,6 +30,8 @@ class AbstractSetting : public Noncopyable, public std::enable_shared_from_this<
 
   virtual const std::string& get() = 0;
 
+  virtual const std::string& get_display_name() {return NULL;}
```

Comment:
Either return "" or the previous string we have used. Don't return NULL.

Reply:
That was maybe a misunderstanding. By returning `NULL` I meant a SQL NULL, it should be the output of the table in case there is no display name. Empty string would also be fine for me.

Reply:
Ok. I will change it to an empty string then 🙃 .

Reply:
Why not default so the string that was used before?

Reply:
The one used before was determined in the python backend, wasn't it? Also fine for me.

Reply:
In the backend, we selected the name column in the `meta_setting` tables and used some string operation to get the last part of the name. That isn't a really nice solution. Here is the code:

https://github.com/hyrise/Cockpit/blob/dev/hyrisecockpit/database_manager/database.py#L286-#L309

So should I return the setting name by default? But then the field in the `display_name` column could have the same value as in the `name` column.

Reply:
We either enforce all plugins to provide a display name (fine with me), or we should have a "somewhat acceptable" default and here I would rather vote for the setting name instead of an empty string. Because empty strings are hard to debug.

So maybe keep it pure virtual and enforce plugins to provide it?

Reply:
We are still talking about **settings** not **plugins**, right? Do we really need to enforce it? We only need the display name for the frontend and not all settings need to be displayed there.

Reply:
I would like to keep it pure virtual, but if we do so isn't that all `settings` need to implement this method `display_name`?

Reply:
If I understand that correctly, yes, if it's a pure virtual method in `AbstractSetting`, all child settings have to implement it.

Reply:
Yes, but I would consider it a good thing to force Plugin developers to think about an understandable and concise setting name.